src/google/protobuf/compiler/ruby/ruby_generated_code_pb.rb

```diff
@@ -70,6 +70,17 @@ module B
     module C
       TestMessage = Google::Protobuf::DescriptorPool.generated_pool.lookup("A.B.C.TestMessage").msgclass
       TestMessage::NestedMessage = Google::Protobuf::DescriptorPool.generated_pool.lookup("A.B.C.TestMessage.NestedMessage").msgclass
+      class TestMessage
+        def optional_enum_const
```

Comment:
I am not a fan of adding these methods by altering the generated Ruby code. I would rather see these new methods added to the object that is returned when calling `lookup("A.B.C.TestMessage").msgclass`. That way, Ruby files that were previously generated will get these new methods by upgrading to the latest google-protobuf gem, without having to regenerate the the Ruby files.